elastic/kibana
# [Maps] Prepare EMSClient for vector tile support

This is a prerequisite for supporting vector tiles from EMS in the Maps app.

-----

- [x] lazily load EMS sub-manifests on a per-service level, instead of all at once
- [x] use EMSClient directly in Maps app
- [x] fix bugs in caching manifest-requests with `_.once`
- [x] rewrite all urls in Kibana-routes to point to proxied endpoints ~~add Kibana proxying logic to the EMSClient (requires routes in Maps app for all EMS endpoints)~~
      - the advantage is that EMSClient now no longer has any embedded proxying logic
- [x] add new test coverage to expanded `EMSClient`

----

To test:
- verify
    - works with `map.proxyElasticMapsServiceInMaps` for `true` and `false`
    - manifests are lazily loaded (e.g. saved map with only rasters should not request vector tile manifest and vice versa)

---

Will create separate PRs for:
   - platform migration so we're using the new http service to make a kibana backend call
   - add endpoints to retrieve vector tile stylesheets, sprite and glypsheets, and mbt tiles
